control.go

```diff
@@ -122,7 +122,7 @@ func hostInfo(addr string, defaultPort int) ([]*HostInfo, error) {
 
 	// Check if host is a literal IP address
 	if ip := net.ParseIP(host); ip != nil {
-		hosts = append(hosts, &HostInfo{hostname: host, connectAddress: ip, port: port})
+		hosts = append(hosts, &HostInfo{hostname: host, connectAddress: ip, port: port, hostId: MustRandomUUID().String()})
```

Comment:
I wonder what is the result of setting to Random here?
This code is used in

```
	hosts, err := addrsToHosts(s.cfg.Hosts, s.cfg.Port, s.logger)
	if err != nil {
		return err
	}
	s.ring.endpoints = hosts
```

I guess the ring is later replaced?

Why can't we use the zero value as a guardian?

Reply:
> > I guess the ring is later replaced?
> 
> I don't think so. As far as I know, gocql doesn't properly separate hosts that it knows about (from system.peers or system.local) and those that are yet to be dialed.
> 
> > Why can't we use the zero value as a guardian?
> 
> If you convert multiple addresses to HostInfos with the zero value, they would overwrite each other in maps where the key is host ID.

Exactly

> I think the proper way would be to have empty HostID and not put hosts whose UUID is not known yet to the ring.

Host discovery can be disabled by `DisableInitialHostLookup` and in such case we won't have any nodes in the ring. Some functions rely on fact that there's at least one node there (several unit tests were panicking inside internal functions when I added suggested logic). I moved random uuids to different place where I can explictly comment why it's needed.